torch/csrc/jit/ir.cpp

```diff
@@ -1651,10 +1661,22 @@ at::ArrayRef<Value*> createTupleUnpack(Value* v) {
 
 std::vector<Value*> inlineCallTo(Node* to_replace, Function* callee) {
   WithInsertPoint guard(to_replace);
+  std::unordered_map<Value*, Value*> value_map;
   auto new_outputs = insertGraph(
       *to_replace->owningGraph(),
       *(callee->optimized_graph()),
-      to_replace->inputs());
+      to_replace->inputs(),
+      value_map);
+
+  // TODO: We might need to use nodes_map instead of value_map. Otherwise, we
+  // are missing nodes without outputs (e.g. prim::Print).
+  std::unordered_set<Node*> updated_nodes;
```

Comment:
I have mixed feeling about this. On one hand, we indeed would move the update code inside the `insertGraph` which will fix the TODO and would allow us to get rid of the `value_map` variant. On the other hand, we would introduce two new arguments (`c10::optional<Function*> callee` and `c10::optional<SourceRange> sr`) which will have overlapping meaning with existing arguments (i.e. "Must `callee->graph()` be the same as `graph`? Should we check that? Should we fail if not?"). It would be ok if we could remove the `graph` argument and replace it with `callee`, but there are some use-cases which insert graph and not the function, so it wouldn't work for them.